modules/http4s/src/smithy4s/http4s/SimpleRestJsonBuilder.scala

```diff
@@ -19,5 +19,7 @@ package http4s
 
 object SimpleRestJsonBuilder
     extends SimpleProtocolBuilder[smithy4s.api.SimpleRestJson](
-      smithy4s.http.json.codecs
+      smithy4s.http.json.codecs(
+        HintMask()
+      ) // TODO smithy4s.api.SimpleRestJson.hintMask??
```

Comment:
> It would be great it we could apply the hint mask across the codecs from the SimpleProtocolBuilder 

We could add a method to the CodecAPI interface tbh.

> I tried looking at the renderer, but couldn't confidently figure out how to begin approaching this 

that's cool, we'll have a brainstorm next week, but the progress at this point in time is already very satisfying.